### PR TITLE
TST: clarify a test where str.strip is called while iterating over file content

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This is a continuation to #16630 and probably the most controversial refactor that emerged from that discussion. Extracted as its own PR to allow merging the non-controversial part of #16630

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Sorry about the noise. I didn't realize the bot would (again) ping everyone. Let me clarify that this PR is only about `config`, `io.ascii` and `io.fits`. All other package labels, as well as `dev-automation`, can be safely removed together with most auto review requests.

**Review by mhvk — COMMENTED** [review]
For the live code, I think sticking with `.strip()` is safer...

File: astropy/io/fits/scripts/fitsdiff.py [hunk 1]
```diff
@@ -80,7 +80,8 @@ def __call__(self, parser, namespace, values, option_string=None):
                 log.warning(f"{self.dest} argument {value} does not exist")
                 return
             try:
-                values = [v.strip() for v in open(value)]
```
[the remark this answers is not in the document]
Comment (dhomeier) [reply astropy/io/fits/scripts/fitsdiff.py]: See https://github.com/astropy/astropy/pull/16630/#discussion_r1659807266 – this is not even reading a fits file or header (so size should be unproblematic), but a list of (as far as I can see) only keywords (card names) to ignore; which might be formatted something like
```
filtnam 
filtnam1
 exptime
```
and the entries read from that are in this branch passed to `setattr` without further stripping.
The doc only specifies _each line in that text file contains one keyword_, so I don't think we can make any further assumptions on the formatting.

File: astropy/io/ascii/qdp.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def _get_lines_from_file(qdp_file):
         lines = qdp_file.split("\n")
     elif isinstance(qdp_file, str):
         with open(qdp_file) as fobj:
-            lines = [line.strip() for line in fobj]
+            lines = fobj.read().splitlines()
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply astropy/io/ascii/qdp.py:141]: I said this a few times already, but for this code, I explicitly **disapprove** unless the `strip()` is kept.

[the remark this answers is not in the document]
Comment (dhomeier) [reply astropy/io/ascii/qdp.py:141]: Those are data files, so I'd rather veto any changes affecting memory footprint or performance for the sake of cleaning up code. Don't know much about the format either, I am afraid there may be no written exact standard – @matteobachetti might know more.

File: astropy/io/fits/scripts/fitsdiff.py [hunk 3]
```diff
@@ -80,7 +80,8 @@ def __call__(self, parser, namespace, values, option_string=None):
                 log.warning(f"{self.dest} argument {value} does not exist")
                 return
             try:
-                values = [v.strip() for v in open(value).readlines()]
+                with open(value) as fd:
+                    values = fd.read().splitlines()
```
Comment (taldcroft) [inline astropy/io/fits/scripts/fitsdiff.py:84]: If this must be changed, please just change this to `values = [v.strip() for v in fd]`. It is NOT worth the out time to determine if the `strip()` is really needed. The original developer wrote it this way for a reason!

File: astropy/io/ascii/qdp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply astropy/io/ascii/qdp.py:141]: > I said this a few times already, but for this code, I explicitly disapprove

Sorry it got lost in the massive discussion over at #16630. I'll revert this one !

File: astropy/io/fits/scripts/fitsdiff.py (hunk 3 above)
Comment (neutrinoceros) [reply astropy/io/fits/scripts/fitsdiff.py:84]: Agreed. So #16630 is sufficient here. I'll revert this one too

**Comment by neutrinoceros** [comment]
only one of the refactors was deemed safe enough to go in so this PR is now only about a single test case in `astropy.config`. Other package labels can be dropped.

File: astropy/config/tests/test_configs.py [hunk 4]
```diff
@@ -352,10 +352,8 @@ class Conf(ConfigNamespace):
     f = tmp_path / "astropy.cfg"
     with open(f, "wb") as fd:
         apycfg.write(fd)
-    with open(f, encoding="utf-8") as fd:
-        lns = [x.strip() for x in fd]
 
-    assert "tstnmo = op2" in lns
+    assert "tstnmo = op2" in f.read_text().splitlines()
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply astropy/config/tests/test_configs.py:356]: Does it matter that it used to be open with utf-8 encoding? was that on purpose?

[the remark this answers is not in the document]
Comment (neutrinoceros) [reply astropy/config/tests/test_configs.py:356]: I believe this was done for portability on Python 2, and it doesn't need to be specified in Python 3 where it's the default behavior of `open` (and `Path.read_text`)

[the remark this answers is not in the document]
Comment (eerovaher) [reply astropy/config/tests/test_configs.py:356]: If the tests still succeed then I'd say the replacement is good enough. But if any evidence should ever emerge that being explicit about the encoding was important then we can specify the encoding in the `read_text()` call.

File: astropy/io/ascii/qdp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matteobachetti) [reply astropy/io/ascii/qdp.py]: I confirm that the written QDP standard is sort of sloppy, and ignored in many places. In general, I would modify these details only if people have an actual issue with real datasets produced by some official tool (e.g. FTOOLS). The current code generally works, with the exception discussed at #15392 where you will find many more details.

**Comment by neutrinoceros** [comment]
Following @eerovaher's request, I moved this back to #16630

**Pull request closed without merging**